Sources/TokamakCore/Environment/EnvironmentObject.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 7/7/20.
+//
+
+import OpenCombine
+
+protocol EnvironmentWriter {
+  func subscribe(_ closure: @escaping () -> ())
+}
+
+@propertyWrapper public struct EnvironmentObject<ObjectType>: EnvironmentReader, EnvironmentWriter where ObjectType: ObservableObject {
```

Comment:
<!--
  1 failure:  Line should be 10...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/TokamakCore/Environment/EnvironmentObject.swift;
  Line: 24;
-->
- :no_entry_sign: Line should be 100 characters or less: currently 135 characters (`line_length`)